Add wpantund benchmark and custom build functionality.

morehouse opened the pull request.

Commit:
Add 1 wpantund fuzzer.

Add wpantund-fuzz but not ncp-spinel-fuzz.  ncp-spinel-fuzz hasn't found
any bugs yet.

Commit:
Add missing dependencies to install-deps.sh.

Commit:
Create simple interface for custom builds.

By including custom-build.sh in a build.sh script, we can now pass
arguments to the build like so:
  ${BENCHMARK}/build.sh [asan|ubsan|hooks] [hooks-file]

asan and ubsan modes build with ASan and UBsan, respectively.  hooks
mode builds with the user-defined hooks in hooks-file.
tutorial/example-hooks.cc provides an example hooks file that simply
dumps coverage information.

Commit:
Allow custom builds on openthread, wpantund, libpng.

Commit:
Add copyright header to custom-build.sh.

Commit:
Add wpantund README.

vitalybuka commented on tutorial/install-deps.sh:
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 sudo apt-get update
-sudo apt-get --yes install curl subversion screen gcc g++ cmake ninja-build golang autoconf libtool apache2 python-dev pkg-config zlib1g-dev libgcrypt11-dev libgss-dev libssl-dev libxml2-dev ragel nasm libarchive-dev make automake
+sudo apt-get --yes install curl subversion screen gcc g++ cmake ninja-build golang autoconf libtool apache2 python-dev pkg-config zlib1g-dev libgcrypt11-dev libgss-dev libssl-dev libxml2-dev ragel nasm libarchive-dev make automake libdbus-1-dev libboost-dev autoconf-archive
```
This is PR to your own branch. It this expected?

morehouse replied:
Yes, each PR depends on the previous one.  AFAIK this is the only way to do this on GitHub.

vitalybuka commented on wpantund-2018-02-27/build.sh:
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Copyright 2018 Google Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Can you please check with out open source docs if this is correct header?

Commit:
Explicitly set C[XX]FLAGS for readabililty.

Setting the flags for the ASan case is not required since the default is
to build with ASan but doing so improves readability.

Commit:
Merge pull request #2 from morehouse/custom_builds

Add custom build functionality.
